libp2p/NodeTable.cpp

```diff
@@ -532,46 +499,22 @@ void NodeTable::onReceived(UDPSocketFace*, bi::udp::endpoint const& _from, bytes
 
 			case PingNode::type:
 			{
-				PingNode in = PingNode::fromBytesConstRef(_from, rlpBytes);
-				if (in.version < dev::p2p::c_protocolVersion)
-				{
-					if (in.version == 3)
-					{
-						compat::Pong p(in.source);
-						p.echo = sha3(rlpBytes);
-						p.sign(m_secret);
-						m_socketPointer->send(p);
-					}
-					else
-						return;
-				}
-				
-				if (RLPXDatagramFace::secondsSinceEpoch() > in.ts)
-				{
-					clog(NodeTableTriviaSummary) << "Received expired PingNode from " << _from.address().to_string() << ":" << _from.port();
-					return;
-				}
-				
+				auto in = dynamic_cast<PingNode const&>(*packet);
 				in.source.address = _from.address();
 				in.source.udpPort = _from.port();
-				addNode(Node(nodeid, in.source));
+				addNode(Node(in.sourceid, in.source));
+				
 				Pong p(in.source);
-				p.echo = sha3(rlpBytes);
+				p.echo = sha3(in.echo);
 				p.sign(m_secret);
 				m_socketPointer->send(p);
 				break;
 			}
-				
-			default:
-				clog(NodeTableWarn) << "Invalid message, " << hex << packetType << ", received from " << _from.address().to_string() << ":" << dec << _from.port();
-				return;
 		}
-
-		noteActiveNode(nodeid, _from);
 	}
-	catch (...)
+	catch (std::exception const& _e)
```

Comment:
This will not catch all exceptions, please either add two catch clauses or remove the additional reason.